newIDE/app/src/MainFrame/ElectronMainMenu.js

```diff
@@ -1,16 +1,31 @@
 // @flow
 import * as React from 'react';
-import MainFrame from '.';
 import optionalRequire from '../Utils/OptionalRequire';
 import { type I18n as I18nType } from '@lingui/core';
 import { t } from '@lingui/macro';
 import { isMacLike } from '../Utils/Platform';
+import { type UpdateStatus } from './UpdaterTools';
 const electron = optionalRequire('electron');
 const ipcRenderer = electron ? electron.ipcRenderer : null;
 
 type Props = {|
-  children: React.Element<typeof MainFrame>,
   i18n: I18nType,
+  project: ?gdProject,
```

Comment:
See my other comment, this looks good but I would move this in a `MainMenuProps`, surely in MainFrame.
If you're worried about MainFrame`/index.js` being bloated (which is kind of true), create a `MainFrame/MainMenu.flow.js` file (actually, this might be better!) (.flow.js is here to indicate that the file contains only types. It's just a convention). Then when you have it, you can import it where needed.